### PR TITLE
Add OperationsPerInvoke to All Benchmarks

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

Add OperationsPerInvoke to All Benchmarks

## Commits

- fix(benchmarks): correct variable names and improve benchmark operations per invoke
- chore(deps): update Microsoft packages to version 10.0.9 in packages.lock.json
- chore(deps): update vm2.TestUtilities to version 2.1.1

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [ ] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -18,7 +18,7 @@
         <PackageVersion Include="xunit.v3.runner.inproc" Version="3.2.2" />
         <PackageVersion Include="FluentAssertions" Version="8.10.0" />
         <PackageVersion Include="NSubstitute" Version="5.3.0" />
-        <PackageVersion Include="vm2.TestUtilities" Version="2.1.0" />
+        <PackageVersion Include="vm2.TestUtilities" Version="2.1.1" />
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply Directory.Packages.props:21]: it's just a chore

File: benchmarks/Ulid/ParseUlid.cs [hunk 2]
```diff
@@ -12,33 +12,58 @@ namespace vm2.Benchmarks.Ulid;
...
 public class ParseUlid
 {
-    const int MaxDataItems = 1000;
-    PreGeneratedData<string> _data2 = null!;
-    PreGeneratedData<byte[]> _data3 = null!;
+    const int operationsPerInvoke = 1000;
+
+    PreGeneratedData<string> _stringData = null!;
+    PreGeneratedData<byte[]> _utf16Data = null!;
 #if GUID_BASELINE
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply benchmarks/Ulid/ParseUlid.cs:19]: renamed

File: benchmarks/Ulid/ParseUlid.cs [hunk 3]
```diff
@@ -12,33 +12,58 @@ namespace vm2.Benchmarks.Ulid;
...
+
+    [Benchmark(Description = "Ulid.Parse(StringUtf8)", OperationsPerInvoke = operationsPerInvoke)]
+    public Ulid Ulid_Parse_Utf8()
+    {
+        Ulid id = default;
 
-    [Benchmark(Description = "Ulid.Parse(StringUtf8)", OperationsPerInvoke = 1000)]
-    public Ulid Ulid_Parse_Utf8() => Ulid.Parse(_data3.GetNext());
+        for (int i = 0; i < operationsPerInvoke; i++)
+            id = Ulid.Parse(_utf16Data.GetNext());
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply benchmarks/Ulid/ParseUlid.cs]: done
